code/modules/projectiles/guns/energy.dm

```diff
@@ -33,9 +36,29 @@
 	shot = ammo_type[select]
 	fire_sound = shot.fire_sound
 	fire_delay = shot.delay
+	SSobj.processing |= src

```

Comment:
Hmmm... Wish there was a less footprinty way to do this while allowing admins to make guns self-charge via VV.